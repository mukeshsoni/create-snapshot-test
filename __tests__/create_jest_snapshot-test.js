const createJestSnapshot = require("../src/index")
const path = require("path")

test("create snapshot test without optional props", () => {
  const fakeComponentPath = path.join(
    __dirname + "/__mocks__/fake_component.js"
  )
  expect(createJestSnapshot(fakeComponentPath)).toMatchSnapshot()
})

test("create snapshot test with optional props", () => {
  const fakeComponentPath = path.join(
    __dirname + "/__mocks__/fake_component.js"
  )
  expect(createJestSnapshot(fakeComponentPath, true)).toMatchSnapshot()
})

test("create snapshot for components without proptypes", () => {
  const fakeComponentPath = path.join(
    __dirname + "/__mocks__/without_proptypes.js"
  )
  expect(createJestSnapshot(fakeComponentPath, true)).toMatchSnapshot()
})

test("create snapshot for components with flow types", () => {
  const fakeComponentPath = path.join(
    __dirname + "/__mocks__/with_flow_types.js"
  )
  expect(createJestSnapshot(fakeComponentPath, true)).toMatchSnapshot()
})
