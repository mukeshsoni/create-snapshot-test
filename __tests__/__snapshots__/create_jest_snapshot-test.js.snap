// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot for components with flow types 1`] = `
"import React from \\"react\\";
import renderer from \\"react-test-renderer\\";
import WithFlowTypes from \\"/Users/mukesh/Documents/projects/create-snapshot-test/__tests__/__mocks__/with_flow_types.js\\";

test(\\"Should render WithFlowTypes correctly\\", () => {
  const tree = renderer.create(<WithFlowTypes name={\\"name\\"} />).toJSON();
  expect(tree).toMatchSnapshot();
});
"
`;

exports[`create snapshot for components without proptypes 1`] = `
"import React from \\"react\\";
import renderer from \\"react-test-renderer\\";
import WithoutProptypes from \\"/Users/mukesh/Documents/projects/create-snapshot-test/__tests__/__mocks__/without_proptypes.js\\";

test(\\"Should render WithoutProptypes correctly\\", () => {
  const tree = renderer.create(<WithoutProptypes />).toJSON();
  expect(tree).toMatchSnapshot();
});
"
`;

exports[`create snapshot test with optional props 1`] = `
"import React from \\"react\\";
import renderer from \\"react-test-renderer\\";
import FakeComponent from \\"/Users/mukesh/Documents/projects/create-snapshot-test/__tests__/__mocks__/fake_component.js\\";

test(\\"Should render FakeComponent correctly\\", () => {
  const tree = renderer
    .create(
      <FakeComponent
        firstName={\\"firstName\\"}
        person={{
          firstName: \\"person.firstName\\",
          lastName: \\"person.lastName\\",
          age: 1
        }}
        hadAadhar={true}
        onClick={function fakeFunction() {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
"
`;

exports[`create snapshot test without optional props 1`] = `
"import React from \\"react\\";
import renderer from \\"react-test-renderer\\";
import FakeComponent from \\"/Users/mukesh/Documents/projects/create-snapshot-test/__tests__/__mocks__/fake_component.js\\";

test(\\"Should render FakeComponent correctly\\", () => {
  const tree = renderer
    .create(
      <FakeComponent
        firstName={\\"firstName\\"}
        person={{
          firstName: \\"person.firstName\\",
          lastName: \\"person.lastName\\",
          age: 1
        }}
        onClick={function fakeFunction() {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
"
`;
